teos/cli/teos_cli.py

```diff
@@ -110,85 +155,267 @@ def stop(self):
+
+    USAGE:  teos-cli get_tower_info
+
+    DESCRIPTION:
+
+        Gets generic information about the tower, like tower_id and aggregate data on users and appointments.
+    """
 
+    name = "get_tower_info"
+
+    @staticmethod
+    def run(rpc_client, opts_args):
+        return rpc_client.get_tower_info()
+
+
+@Cli._register_command
+class GetUsersCommand(CliCommand):
+    """
+    NAME:   teos-cli get_users - Gets the list of registered user ids.
+
+    USAGE:  teos-cli get_users
+
+    DESCRIPTION:
+
+        Gets an array with the user ids of all the users registered to the tower.
+    """
+
+    name = "get_users"
+
+    @staticmethod
+    def run(rpc_client, opts_args):
+        return rpc_client.get_users()
+
+
+@Cli._register_command
+class GetUserCommand(CliCommand):
+    """
+    NAME:   teos-cli get_user - Gets information about a specific user.
+
+    USAGE:  teos-cli get_user "user_id"
+
+    DESCRIPTION:
+
+        Gets information about a specific user.
+    """
+
+    name = "get_user"
+
+    @staticmethod
+    def run(rpc_client, opts_args):
+        opts, args = opts_args
+
+        if not args:
+            sys.exit("No user_id was given")
+        if len(args) > 1:
+            sys.exit(f"Expected only one argument, not {len(args)}")
+
+        return rpc_client.get_user(args[0])
+
+
+@Cli._register_command
+class StopCommand(CliCommand):
+    """
+    NAME:   teos-cli stop - Requests a graceful shutdown of the tower..
+
+    USAGE:  teos-cli stop
+
+    DESCRIPTION:
+
+        Requests a graceful shutdown of the tower.
+    """
+
+    name = "stop"
+
+    @staticmethod
+    def run(rpc_client, opts_args):
+        rpc_client.stop()
+        print("Closing the Eye of Satoshi")
+
+
+@Cli._register_command
+class HelpCommand(CliCommand):
+    """
+    NAME:   teos-cli help - Shows general help, or help for a specific command.
+
+    USAGE:  teos-cli help [command]
+
+    DESCRIPTION:
+
+        Shows a summary of all the commands and global options. If command is given, shows detailed help for the
+        command.
+    """
+
+    name = "help"
+
+    @staticmethod
+    def run(rpc_client, opts_args):
+        opts, args = opts_args
+
+        print(opts_args)
+
+        if not args:
+            sys.exit(show_usage())
+        elif len(args) > 1:
+            sys.exit(f"Expected only one argument, not {len(args)}")
+
+        command = args.pop(0)
+        if command not in Cli.COMMANDS:
+            sys.exit("Unknown command.")
+
+        sys.exit(Cli.COMMANDS[command].__doc__)
+
+
+def main():
+    # Split the command line as follows: global options, command, command args,
+    # where command is the first argument not starting with "-".
+
+    try:
+        cmd_index = next(i for i in range(len(argv)) if i > 0 and not argv[i].startswith("-"))
+    except StopIteration:
+        sys.exit(show_usage())
+
+    command = argv[cmd_index]
+    global_options = argv[1:cmd_index]
+    command_args = argv[cmd_index + 1 :]
```

Comment:
I don't think this covers all the cases.

What if there is a command that has multiple arguments and also command options?

Reply:
Seems good to me. Do you have a failure example?
My assumption here is that the format is:
```
  teos-cli [global_options] command [command options and/or arguments]
```
where all the global_options must start with 1 or two `-`, and `command` definitely does not.
Since `cmd_index` is the first command line arg not starting with `-`, and the parsing of everything after `command` is delegated to the `CLICommand` itself, this seems sound to me.

Reply:
Oh OK, so whatever comes after the command is passed to the specific command for custom parsing, right?

Reply:
Yeah, that's the idea, the Command's `parse_args` will take care of the rest.

Reply:
OK, will need to do a bit of testing but looks good